File: src/main/java/com/kingdom/util/CardNameComparator.java
package com.kingdom.util;

import com.kingdom.model.Card;

import java.util.Comparator;

public class CardNameComparator implements Comparator<Card> {
    public int compare(Card c1, Card c2) {
        if (c1.getActionValue() == c2.getActionValue()) {
            return c1.getName().compareTo(c2.getName());
        } else {
            return Integer.valueOf(c1.getActionValue()).compareTo(c2.getActionValue());
        }
    }
}

File: src/main/kotlin/com/kingdom/util/CardNameComparator.kt
package com.kingdom.util

import com.kingdom.model.cards.Card

import java.util.Comparator

class CardNameComparator : Comparator<Card> {
    override fun compare(c1: Card, c2: Card): Int {
        return if (c1.actionValue == c2.actionValue) {
            c1.name.compareTo(c2.name)
        } else {
            Integer.valueOf(c1.actionValue)!!.compareTo(c2.actionValue)
        }
    }
}
